modules/system/assets/js/framework.js

```diff
@@ -99,10 +99,48 @@ if (window.jQuery.request !== undefined) {
 
                 delete data[inputName]
             }
+            
+            /*
+             * JavaScript Object to FormData, with support for nested objects, arrays and File objects.
+             * https://gist.github.com/ghinda/8442a57f22099bdb2e34
+             */
+            var objectToFormData = function(obj, form, namespace) {
+
+                var fd = form || new FormData();
+                var formKey;
+
+                for(var property in obj) {
+                    if(obj.hasOwnProperty(property)) {
+
+                        if(namespace) {
+                            formKey = namespace + '[' + property + ']';
+                        } else {
+                            formKey = property;
+                        }
+
+                        // if the property is an object, but not a File,
+                        // use recursivity.
+                        if(typeof obj[property] === 'object' && !(obj[property] instanceof File)) {
+
+                            objectToFormData(obj[property], fd, property);
 
-            $.each(data, function(key) {
+                        } else {
+
+                            // if it's a string or a File object
```

Comment:
This comment is not technically correct, it would be more accurate to say "if it's a File or not an object".